policy-controller/k8s/index/src/index.rs

```diff
@@ -960,7 +967,7 @@ impl Pod {
 
     /// Updates a pod-port to use the given named server.
     fn set_default_server(&mut self, port: NonZeroU16, config: &ClusterInfo) {
-        let server = Self::default_inbound_server(port, &self.meta.settings, config);
+        let server = Self::default_inbound_server(port, &self.meta.settings, &self.probes, config);
```

Comment:
This won't work because it's called while there's already a borrow on `self`: https://github.com/linkerd/linkerd2/blob/ae974aa1ea8590eb9a8dea2681d75b6301707d61/policy-controller/k8s/index/src/index.rs#L1015-L1019

Reply:
Yea unfortunately I tried to make this change, but as Oliver already pointed out we've already borrowed as `&mut self` and can't pass it in to another method.